### PR TITLE
KEYCLOAK-15806: Extension to SignatureAlgorithm to support more Algorithms

### PR DESCRIPTION
**Pull request by rowe42**
Extension to SignatureAlgorithm to support more Algorithms (RSA_SHA256_MGF1, RSA_SHA512_MGF1).
    Also included in realms.js so it can be chosen as signature algorithm when brokering through SAML.

### CODE REVIEW

File: themes/src/main/resources/theme/base/admin/resources/js/controllers/realm.js [hunk 1]
```diff
@@ -841,7 +841,9 @@ module.controller('RealmIdentityProviderCtrl', function($scope, $filter, $upload
             "RSA_SHA1",
             "RSA_SHA256",
             "RSA_SHA512",
-            "DSA_SHA1"
+            "DSA_SHA1",
+            "RSA_SHA256_MGF1",
+            "RSA_SHA512_MGF1"
```
Comment (hmlnarik) [inline themes/src/main/resources/theme/base/admin/resources/js/controllers/realm.js:846]: These changes should be done consistently also for SAML clients.

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank you for the PR. Please see inline.

**Comment by rowe42** [comment]
fixed. sorry for the stupid copy-paste-error!

**Comment by hmlnarik** [comment]
Thank you. For some reason setting the client signature algorithm to one of the MGF variants, the client does not display the signature algorithm in client detail. Could you please check?

Also it would be great if the new algorithms were placed next to their non-MGF variant in the combo box, currently they are at the end of the list.

**Comment by rowe42** [comment]
@hmlnarik You were right. There was some code missing in clients.js, which takes care of setting the chosen entry to the data model.

I've fixed that now and rearranged the entries as suggested.

**Review by hmlnarik — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**